### DISCUSSION TITLE
Different colors for functions, user defined functions, macros, and variables.

### DISCUSSION THREAD

**Discussion by ghost** [genius257/vscode-autoit#57] [Ideas]
Currently functions, UDF and variables are all white.
It would be amazing if they could each have their own color.
also some AutoIt operators appear as white too (such as not, or)

**Comment by genius257** [comment]
Hi @danielbarnes,

The colors are currently implemented by default colors for the known types via rule name in the [au3.tmLanguage.json](https://github.com/genius257/vscode-autoit/blob/master/syntaxes/au3.tmLanguage.json).

For example, operators, like not and or are set as keywords:
https://github.com/genius257/vscode-autoit/blob/d4b2f97ed7b327aed9e30f9c7afd22dbbc6c2b7b/syntaxes/au3.tmLanguage.json#L58-L60
Whereas function names are set as entity function names:
https://github.com/genius257/vscode-autoit/blob/d4b2f97ed7b327aed9e30f9c7afd22dbbc6c2b7b/syntaxes/au3.tmLanguage.json#L66-L69

I know that coloring can depend from theme to theme. In my case (Dark Modern) function names and variables have different colors:
![image](https://github.com/user-attachments/assets/26c5f866-fe2a-4107-8da1-5749c56afd20)


I found some vscode documentation you could use to customize colors via settings (JSON):
https://code.visualstudio.com/docs/getstarted/themes#_editor-syntax-highlighting

I wrote a small example with horrible colors, to demonstrate:
```json
{
    "editor.tokenColorCustomizations": {
        "functions": "#FF0000",
        "keywords": "#00FF00",
        "textMateRules": [
            {
                "scope": "keyword.operator",
                "settings": {
                    "foreground": "#FFFF00"
                }
            }
        ]
    }
}
```
To make this only affect AutoIt3 files, you could wrap it in:
```json
"[au3]": {
     // insert custom settings for au3 only here...
},
```

In regards to UDF and functions being the same color, i assume you mean that you would like different colors for the builtin functions vs user defined functions?

Currently that is not something i have planned, and am not sure if should be 😅 To support picking a target au3 version, it would haft to be implemented on the language server, and it is not ready for performance critical things like coloring, yet. 😄
